code/game/turfs/simulated/floor.dm

```diff
@@ -89,6 +90,9 @@ var/list/icons_to_ignore_at_floor_init = list("damaged1","damaged2","damaged3","
 /turf/simulated/floor/proc/update_icon()
 	if(air)
 		update_visuals()
+	overlays.Cut(current_overlay)

```

Comment:
`overlays -= current_overlay` accomplishes the same thing and is easier to understand for someone familiar with our codebase.